File: arpg_prg/server_prg/src/com/game/core/GameAction.java
package com.game.core;

import io.netty.channel.ChannelHandlerContext;

public abstract class GameAction
{
	protected GameAction (int actionId)
	{
		this.actionId = actionId;
	}
	
	public abstract void GetRequestMsg(ChannelHandlerContext ctx, int messageID, byte[] array);
	public abstract void PutRespondMsg(Object userdata);

	protected int actionId;
	protected ChannelHandlerContext ctx;
}

File: arpg_prg/Fantasy/Assets/Code/Core/NetWork/GameAction.cs
using System;

namespace Net
{
	public abstract class GameAction
	{
		protected GameAction (int actionId)
		{
			_actionId = actionId;
			ActionFactory.Instance.RegisterAction (actionId, this);
		}
			
		public abstract byte[] GetRequestMsg(object userdata = null);
		public abstract void PutRespondMsg(byte[] msg, object userdata = null);

		private int _actionId;
		public int ActionId { get { return _actionId; } }
	}
}

